brat/Data/Hugr.hs

```diff
@@ -192,44 +238,25 @@ instance ToJSON node => ToJSON (FuncDefn node) where
                                     ,"signature" .= signature_
                                     ]
 
-data HugrConst
-  = HCInt Int
-  | HCFloat Double
-  | HCUnit
-  | HCFunction (Hugr Int)
- deriving (Eq, Show)
+data CustomConst where
+  CC :: forall a. (Eq a, Show a, ToJSON a) => String -> a -> CustomConst
 
-instance ToJSON HugrConst where
-  toJSON (HCInt x) = object ["v" .= ("Extension" :: Text)
-                            ,"c" .= [object ["c" .= ("ConstIntS" :: Text)
-                                            ,"log_width" .= (intWidth :: Int)
-                                            ,"value" .= x
-                                            ]
-                                    ]
-                            ]
-  toJSON (HCFloat x) = object ["v" .= ("Extension" :: Text)
-                              ,"c" .= [object ["c" .= ("ConstF64" :: Text)
-                                              ,"value" .= x
-                                              ]
-                                      ]
-                              ]
-  toJSON HCUnit = object ["v" .= ("Tuple" :: Text)
-                         ,"vs" .= ([] :: [()])
-                         ]
-  toJSON (HCFunction hugr) = object ["v" .= ("Function" :: Text)
-                                    ,"hugr" .= hugr
-                                    ]
+instance Eq CustomConst where
+  _ == _ = False
```

Comment:
This is a bit sus. I presume we can't write a proper `Eq` because `a` is quantified?

Reply:
I suppose with the keymap change we could!